src/layers/mpls.rs

```diff
@@ -0,0 +1,72 @@
+//! Multi Label Protocol Switching
+
+use std::convert::TryInto;
+
+use serde::Serialize;
+
+use crate::errors::Error;
+use crate::layers::ethernet;
+use crate::types::{ETHERTYPE_MPLS_MULTICAST, ETHERTYPE_MPLS_UNICAST};
+use crate::Layer;
+
+use super::ipv4::IPv4;
+
+/// Default Header Length for MPLS Packets
+pub const MPLS_HDR_LEN: usize = 4_usize;
+
+// Register Ourselves to the Ethernet layer, as this is a 2.5 layer protocol
+pub(crate) fn register_defaults() -> Result<(), Error> {
+    ethernet::register_ethertype(ETHERTYPE_MPLS_UNICAST, MPLS::creator)?;
+    ethernet::register_ethertype(ETHERTYPE_MPLS_MULTICAST, MPLS::creator)
+}
+
+#[derive(Debug, Default, Serialize)]
+pub struct MPLS {
```

Comment:
This structure should support a label stack. A label stack can be simply implemented by having something like `vec<MPLSLabel>` and then this structure should look like 

```rust
struct MPLS { 
    labels: vec<MPLSLabel>
}
```